packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -207,7 +207,14 @@ abstract class SchedulerBinding extends BindingBase {
   void scheduleTask(VoidCallback task, Priority priority) {
     final bool isFirstTask = _taskQueue.isEmpty;
     _taskQueue.add(new _TaskEntry(task, priority.value));
-    if (isFirstTask)
+    if (isFirstTask && !locked)
+      _ensureEventLoopCallback();
```

Comment:
seems like you did it in the other file, perhaps assert !locked in _ensureEventLoopCallback too